File: 2012/December/05-12-2012/spec/RomanNumeralSpec.js
describe("RomanNumeral", function() {

  beforeEach(function(){
    romanNumeral = new RomanNumeral();
  })

  it('should return the roman numeral for 1', function() {
    expect(romanNumeral.convert(1)).toEqual("I");
  })

  it('should return the roman numeral for 5', function() {
    expect(romanNumeral.convert(5)).toEqual("V");
  })

  it('should return the roman numeral for 10', function() {
    expect(romanNumeral.convert(10)).toEqual("X");
  })

  it('should return the roman numeral for 50', function() {
    expect(romanNumeral.convert(50)).toEqual("L");
  })

  it('should return the roman numeral for 100', function() {
    expect(romanNumeral.convert(100)).toEqual("C");
  })

  it('should return the roman numeral for 500', function() {
    expect(romanNumeral.convert(500)).toEqual("D");
  })

  it('should return the roman numeral for 1000', function() {
    expect(romanNumeral.convert(1000)).toEqual("M");
  })

  it('should return the roman numeral for 55', function() {
    expect(romanNumeral.convert(55)).toEqual("LV");
  })

})

File: 2012/December/13-12-2012/roman_numeral_spec.rb
require_relative 'roman_numeral'

describe 'RomanNumeral' do
  let(:numeral) {RomanNumeral.new}
  it "converts 1 to I" do
    numeral.toRoman(1).should eq "I"
  end

  it "converts 2 to II" do
    numeral.toRoman(2).should eq "II"
  end

  it "converts 3 to III" do
    numeral.toRoman(3).should eq "III"
  end

  it "converts 4 to IV" do
    numeral.toRoman(4).should eq "IV"
  end

  it "converts 5 to V" do
    numeral.toRoman(5).should eq "V"
  end

  it "converts 6 to VI" do
    numeral.toRoman(6).should eq "VI"
  end

  it "converts 140 to CXL" do
    numeral.toRoman(140).should eq "CXL"
  end

  it "converts 2990 to MMCMXC" do
    numeral.toRoman(2990).should eq "MMCMXC"
  end

  it "converts 4323 to MMMMCCCXXIII" do
    numeral.toRoman(4323).should eq "MMMMCCCXXIII"
  end
end